From: Stefano Rivera <stefanor@debian.org>
Date: Sat, 4 Nov 2023 18:59:01 +0200
Subject: Use virtualenvs' bundled wheel and setuptools

We can't install it from the Internet in our builds, without Internet
access.

Bug-Debian: https://bugs.debian.org/1055337
---
 testing/cffi0/test_zintegration.py | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/testing/cffi0/test_zintegration.py b/testing/cffi0/test_zintegration.py
index 9fbbd91..53f48eb 100644
--- a/testing/cffi0/test_zintegration.py
+++ b/testing/cffi0/test_zintegration.py
@@ -21,21 +21,9 @@ def create_venv(name):
             #'--never-download', <= could be added, but causes failures
             # in random cases on random machines
                                '-p', os.path.abspath(sys.executable),
+                               '--wheel=bundle', '--setuptools=bundle',
                                str(tmpdir)])
 
-        # Python 3.12 venv/virtualenv no longer include setuptools and wheel by default, which
-        # breaks a number of these tests; ensure it's always present for 3.12+
-        if sys.version_info >= (3, 12):
-            subprocess.check_call([
-                os.path.join(tmpdir, 'bin/python'),
-                '-m',
-                'pip',
-                'install',
-                'setuptools',
-                'wheel',
-                '--upgrade'
-            ])
-
     except OSError as e:
         pytest.skip("Cannot execute virtualenv: %s" % (e,))
 
